services/billing.py

```diff
@@ -155,18 +157,75 @@ def delete_subscription(self, owner: Owner):
             f"Downgrade to basic plan from user plan for owner {owner.ownerid} by user #{self.requesting_user.ownerid}",
             extra=dict(ownerid=owner.ownerid),
         )
+
         if subscription_schedule_id:
             log.info(
                 f"Releasing subscription from schedule for owner {owner.ownerid} by user #{self.requesting_user.ownerid}",
                 extra=dict(ownerid=owner.ownerid),
             )
             stripe.SubscriptionSchedule.release(subscription_schedule_id)
 
-        stripe.Subscription.modify(
-            owner.stripe_subscription_id,
-            cancel_at_period_end=True,
-            proration_behavior="none",
+        # we give an auto-refund grace period of 24 hours for a monthly subscription or 72 hours for a yearly subscription
+        current_subscription_datetime = datetime.fromtimestamp(
```

Comment:
Just to confirm, are the timezones for both datetime.now() and datetime.fromtimestamp() the same?

Reply:
That's a great question. I just checked and both are in machine local so since we only care about the difference, we're good.

Reply:
great! Thanks for checking

Reply:
I'm converting these to utc now actually because I realized depending on what the local time of the machine is, it could change the date. Like if february 29th was converted to march 1st, subtracting 1 month would give february 1st instead of january 29th